packages/react-sdk/src/client.ts

```diff
@@ -0,0 +1,595 @@
+      overrideAttributes,
+    )
+    if (userId === null) {
+      return null
+    }
+    return this.client.getFeatureVariableInteger(feature, variable, userId, attributes)
+  }
+
+  /**
+   * @param {string} feature
+   * @param {string} variable
+   * @param {string} userId
+   * @param {UserAttributes} [attributes]
+   * @returns {(number | null)}
+   * @memberof OptimizelySDKWrapper
+   */
+  public getFeatureVariableDouble(
+    feature: string,
+    variable: string,
+    overrideUserId?: string,
+    overrideAttributes?: UserAttributes,
+  ): number | null {
+    const [userId, attributes] = this.getUserIdAndAttributes(
+      overrideUserId,
+      overrideAttributes,
+    )
+    if (userId === null) {
+      return null
+    }
+    return this.client.getFeatureVariableDouble(feature, variable, userId, attributes)
+  }
+
+  /**
+   * Get an array of all enabled features
+   *
+   * @param {string} userId
+   * @param {UserAttributes} [attributes]
+   * @returns {Array<string>}
+   * @memberof OptimizelySDKWrapper
+   */
+  public getEnabledFeatures(
+    overrideUserId?: string,
+    overrideAttributes?: UserAttributes,
+  ): Array<string> {
+    const [userId, attributes] = this.getUserIdAndAttributes(
+      overrideUserId,
+      overrideAttributes,
+    )
+    if (userId === null) {
+      return []
+    }
+    return this.client.getEnabledFeatures(userId, attributes)
+  }
+
+  /**
+   * @param {string} experiment
+   * @param {string} userId
+   * @returns {(string | null)}
+   * @memberof OptimizelySDKWrapper
+   */
+  public getForcedVariation(experiment: string, overrideUserId?: string): string | null {
+    const [userId, _] = this.getUserIdAndAttributes(overrideUserId)
+    if (userId === null) {
+      return null
+    }
+    return this.client.getForcedVariation(experiment, userId)
+  }
+
+  /**
+   * @param {string} experiment
+   * @param {string} userId
+   * @param {string} variationKey
+   * @returns {boolean}
+   * @memberof OptimizelySDKWrapper
+   */
+  public setForcedVariation(
+    experiment: string,
+    overrideUserIdOrVariationKey: string,
+    variationKey?: string | null,
+  ): boolean {
+    let finalUserId: string | null = null
+    let finalVariationKey: string | null = null
+    if (arguments.length === 2) {
+      finalVariationKey = overrideUserIdOrVariationKey
+      finalUserId = this.getUserIdAndAttributes()[0]
+    } else if (arguments.length === 3) {
+      const res = this.getUserIdAndAttributes(overrideUserIdOrVariationKey)
+      finalUserId = res[0]
+      if (variationKey === undefined) {
+        // can't have undefined if supplying all 3 arguments
+        return false
+      }
+      finalVariationKey = variationKey
+    }
+
+    if (finalUserId === null) {
+      return false
+    }
+    return this.client.setForcedVariation(experiment, finalUserId, finalVariationKey)
+  }
+
+  public close() {
+    return this.client.close()
+  }
+
+  protected getUserIdAndAttributes(
```

Comment:
I think we should name this something like `getUserContext` and return a proper `UserContext` object so that we don't limit ourselves to just ID and attributes.